InAppPurchasesAtlasAppServices/FoodieFolio/FoodieFolio/Managers/StoreKitManager.swift

```diff
@@ -0,0 +1,149 @@
+           let plistDict = try? PropertyListSerialization.propertyList(from: plist, format: nil) as? [String: String] {
+            productDict = plistDict
+        } else {
+            productDict = [:]
+        }
+
+        updateListenerTask = transactionListener()
+
+        // Fetch products asynchronously when the StoreKitManager is initialized
+        Task {
+            await requestProducts()
+
+            await updateCustomerProductStatus()
+        }
+    }
+
+    // If user quits app for example, stop listening for transactions
+    deinit {
+        updateListenerTask?.cancel()
+    }
+
+    func transactionListener() -> Task<Void, Error> {
+        return Task.detached {
+            // Iterate through any transactions that don't come from a direct call to purchase() method
+            for await result in Transaction.updates {
+                do {
+                    let transaction = try self.checkVerified(result)
+
+                    // transaction is verified, so the content unlocks for the user
+                    await self.updateCustomerProductStatus()
+
+                    // The transaction always needs to be finished
+                    await transaction.finish()
+                } catch {
+                    print("Transaction failed verification")
+                }
+            }
+        }
+    }
+
+    /// Fetches available products asynchronously and updates `storeProducts`.
+    @MainActor
+    func requestProducts() async {
+        do {
+            // Fetch products using productDict values (IDs)
+            storeProducts = try await Product.products(for: productDict.values)
+        } catch {
+            // Handle errors if product fetching fails
+            print("Failed - error retrieving products \(error)")
+        }
+    }
+
+    /// Call product purchase and returns an optional transaction
+    func purchase(_ product: Product) async throws -> Transaction? {
+        let result = try await product.purchase()
+
+        switch result {
+        case .success(let verificationResult):
+            // Transaction will be verified for automatically using JWT
+            let transaction = try checkVerified(verificationResult)
+
+            await updateCustomerProductStatus()
+
+            await transaction.finish()
+
+            return transaction
+        case .userCancelled, .pending:
+            return nil
+        default:
+            return nil
+        }
+    }
+
+    func checkVerified<T>(_ result: VerificationResult<T>) throws -> T {
+        // Check if JWS passes the StoreKit verification
+        switch result {
+        case .unverified:
+            // failed verification
+            throw StoreError.failedVerification
+        case .verified(let signedType):
+            // Result is verified, return the unwrapped value
+            return signedType
+        }
+    }
+
+    @MainActor
+    func updateCustomerProductStatus() async {
```

Comment:
Any reason to do this in the @MainActor

Reply:
Not that I can remember, I think I was doing some tests and forgot to remove it